### ISSUE TITLE
GDK/Cairo failures when taking a screenshot

### ISSUE THREAD

**Issue by nenad** [jtheoof/swappy#53]
`grim -g "$(slurp)" - | swappy -f -`

Produces the following output when I try to save the file: 
```bash
(swappy:3089974): Gdk-CRITICAL **: 18:42:54.734: gdk_pixbuf_get_from_surface: assertion 'width > 0 && height > 0' failed

(swappy:3089974): GdkPixbuf-CRITICAL **: 18:42:54.734: gdk_pixbuf_savev: assertion 'GDK_IS_PIXBUF (pixbuf)' failed

(swappy:3089974): GLib-GObject-CRITICAL **: 18:42:54.736: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
```
I get a notification that the file has been saved, but it doesn't exit on the disk.

And `swappy -f hello.png` fails with:
```bash
** (swappy:3077837): WARNING **: 18:38:36.490: error while loading png file: hello.png - cairo status: file not found
```

`swappy --version` -> `swappy version 1.2.1-8e5b959 (Aug  3 2020, branch 'master')`

I run Arch Linux and I have gtk3, glib2 and glibc installed.

**Comment by jtheoof** [comment]
That's weird. I can't reproduce this issue and it's the first time it's come up. I will need more details to investigate this. Version of sway, a little more details of what's happening when calling: `pixbuf_get_from_state`. It's as if the rendered surface was absent at that time.

**Comment by nenad** [comment]
Okay, I found the problem. Turns out, I have set the `GDK_BACKEND` environment variable to `x11` when I launch my terminal (termite). Every child process after that inherited the environment and I suppose that doesn't play well with GTK/Cairo. When explicitly set to `wayland` it worked. Interestingly enough, all of the functionality works, except for the saving part. I think this issue can be closed. I'm not sure if swappy should be checking for this env var, or that might be out of the scope of the project.
